### PR TITLE
added dynamic header menu, hugo-style

### PR DESCRIPTION
**Pull request by gavinr**
see #27

### CODE REVIEW

**Comment by EmielH** [comment]
Thanks a lot for the PR! I'm a bit busy at the moment, I'll check it out as soon as possible.

File: layouts/partials/header-menu.html [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+                    <span>{{ .Name }}</span>
+                </a>
+            </li>
+            <ul class="sub-menu">
+                {{ range .Children }}
+                    <li class="{{ if $currentPage.IsMenuCurrent "main" . }}active{{ end }}">
+                        <a href="{{ .URL }}">{{ .Name }}</a>
+                    </li>
+                {{ end }}
+            </ul>
```
Comment (EmielH) [inline layouts/partials/header-menu.html]: Creating submenus looks a bit weird. For example, add the following to your config file:

```
[menu]
  [[menu.main]]
    identifier = "about"
    name = "About"
    title = "About"
    url = "/about/"
    weight = 0
  [[menu.main]]
    identifier = "posts"
    name = "Posts"
    title = "Posts"
    url = "/posts/"
    weight = 0
  [[menu.main]]
    identifier = "topic"
    parent = "posts"
    name = "Topic"
    title = "Topic"
    url = "/posts/topic/"
    weight = 0
  [[menu.main]]
    identifier = "anothertopic"
    parent = "posts"
    name = "Another topic"
    title = "Another topic"
    url = "/posts/anothertopic/"
    weight = 0
```

I don't think the theme is very well suited for nested menu structures at the moment, so I would suggest to remove this part.

The CSS class `active` is also not used in Tale, but removing this entire section already resolves that issue.

File: layouts/partials/header-menu.html [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+        {{ else }}
+            <li>
+                <a href="{{ .URL }}">
+                    {{ .Pre }}
```
Comment (EmielH) [inline layouts/partials/header-menu.html]: If we include `.Pre` we should probably include `.Post` as well. See the [list of menu entry variables](https://gohugo.io/variables/menus/#menu-entry-variables) (mainly leaving this here as future reference for myself 🙂).

File: layouts/partials/header-menu.html [hunk 3]
```diff
@@ -1,4 +1,27 @@
 <ul>
-    <li><a href="{{ "about" | relURL }}">{{ i18n "about" }}</a></li>
-    <li><a href="{{ "/" | relURL }}">{{ i18n "posts" }}</a></li>
```
Comment (EmielH) [inline layouts/partials/header-menu.html:3]: These i18n entries can now be removed from all the i18n files as well.

File: layouts/partials/header-menu.html [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+                    <li class="{{ if $currentPage.IsMenuCurrent "main" . }}active{{ end }}">
+                        <a href="{{ .URL }}">{{ .Name }}</a>
+                    </li>
+                {{ end }}
+            </ul>
+        {{ else }}
+            <li>
+                <a href="{{ .URL }}">
+                    {{ .Pre }}
+                    <span>{{ .Name }}</span>
```
Comment (EmielH) [inline layouts/partials/header-menu.html]: The added `<span>` differs from the original HTML output, but I don't really mind. It separates the `.Pre` and `.Post` variables from the link name.

**Review by EmielH — CHANGES_REQUESTED** [review]
Thanks a lot for looking into this and submitting this PR. I think this is a great feature! 👍  My main concern is that Tale doesn't seem well-suited for nested menus at the moment. I think having a one level menu suffices for now. Could you check my review? Thanks a lot!

File: layouts/partials/header-menu.html (hunk 3 above)
Comment (gavinr) [reply layouts/partials/header-menu.html:3]: Removed.

File: layouts/partials/header-menu.html (hunk 4 above)
Comment (gavinr) [reply layouts/partials/header-menu.html]: Sounds good!

File: layouts/partials/header-menu.html (hunk 2 above)
Comment (gavinr) [reply layouts/partials/header-menu.html]: Added `.Post`.

File: layouts/partials/header-menu.html (hunk 1 above)
Comment (gavinr) [reply layouts/partials/header-menu.html]: Removed submenu.

**Review by EmielH — APPROVED** [review]
Thanks a lot for your help!

**Comment by gavinr** [comment]
> 
> 
> Thanks a lot for your help!

No problem! Thank you for maintaining this project!

**Pull request merged**